### PR TITLE
Upload caching files to AWS S3 bucket

### PR DESCRIPTION
**Pull request by codemagic-mohammed**

### CODE REVIEW

File: content/knowledge-others/install-unity-version.md [hunk 1]
```diff
@@ -99,32 +99,93 @@ Use the Unity version you installed on the machine:
...
+When building with the new Unity version, it's better to add the **-buildTarget** argument to set the build target to `android` or `ios`.
+{{</notebox>}}
 ## Caching
-You can cache the newly installed version on Unity to avoid installing it in every build, read more about caching [here](https://docs.codemagic.io/yaml/yaml-getting-started/#cache).
+Instead of waiting from 5 to 7 minutes in every build to download and install your Unity editor, you can cache it using your external storage tool like AWS S3 or GCP.
+
+In the following example we are going to use AWS S3 to store our cached files.
+
+#### Using AWS S3
+In order to use AWS S3, you need to configure your access credentials in Codemagic. You can follow the [instructions](https://aws.amazon.com/getting-started/hands-on/backup-to-s3-cli/) provided by Amazon to create your account and get the necessary details.
```
Comment (nevercode-kevin) [inline content/knowledge-others/install-unity-version.md:117]:
```suggestion
In order to use AWS S3, you need to configure your access credentials in Codemagic. You can follow the [instructions](https://aws.amazon.com/getting-started/hands-on/backup-to-s3-cli/) provided by Amazon to create your account and get the necessary credentials.
```

File: content/knowledge-others/install-unity-version.md [hunk 2]
```diff
@@ -99,32 +99,93 @@ Use the Unity version you installed on the machine:
...
-    cache:
-      cache_paths:
-        - /Applications/Unity/Hub/Editor/${UNITY_VERSION}
+environment:
+  groups:
+    - aws_credentials
+scripts:
+  - name: Install awscli
+    script: | 
+      sudo pip3 install awscli --upgrade
```
Comment (nevercode-kevin) [inline content/knowledge-others/install-unity-version.md:136]: AWS cli 2.8.9 is pre-installed, so shouldn't need to be upgraded for S3 uploads.

File: content/knowledge-others/install-unity-version.md [hunk 3]
```diff
@@ -99,32 +99,93 @@ Use the Unity version you installed on the machine:
...
+  script: | 
+    if [ "$S3_HAS_EDITOR" == "false" ] ; then
+      cd /Applications/Unity/Hub/Editor/
+      tar -cv ${UNITY_VERSION}/ | gzip > ${UNITY_VERSION}.tar.gz
+    fi
+- name: Copy editor to S3 bucket
+  script: | 
+    if [ "$S3_HAS_EDITOR" == "false" ] ; then
+      aws s3 cp /Applications/Unity/Hub/Editor/${UNITY_VERSION}.tar.gz s3://<BUCKET_NAME>
+    fi
```
Comment (nevercode-kevin) [inline content/knowledge-others/install-unity-version.md:177]: Should this be done at the end of the workflow, in the Publishing section scripts, if required?

**Review by nevercode-kevin — CHANGES_REQUESTED** [review]

**Review by kevin-suhajda — APPROVED** [review]
LGTM

**Pull request merged**